### PR TITLE
Replace glob_match with globset

### PR DESCRIPTION
**Pull request by jacobotb**
## What problem are you trying to solve?
The `glob_match` crate has a bug with a pattern like `**/*test*.py` , where a file like `abc/test/def/test.py` does not match.

Or even `t/test.py` does not match.

See https://github.com/devongovett/glob-match/issues/9 for issue.

## What is your solution?
The `globset` crate does not have that bug and is pretty fast too.

## What the reviewer should know
`globset` has a separate compilation step, so I do it during configuration parsing.

### CODE REVIEW

File: crates/bins/src/bin/datadog-static-analyzer.rs [hunk 1]
```diff
@@ -55,10 +55,20 @@ fn print_configuration(configuration: &CliConfiguration) {
...
+            .collect::<Vec<String>>()
+            .join(",")
     };
     let only_paths_str = match &configuration.path_config.only {
-        Some(x) => x.join(","),
+        Some(x) => x
+            .iter()
+            .map(|p| p.clone().into())
+            .collect::<Vec<String>>()
+            .join(","),
```
Comment (juli1) [inline crates/bins/src/bin/datadog-static-analyzer.rs:71]: Curious why this change? Seems more complicated to read?

File: crates/bins/src/bin/datadog-static-analyzer.rs [hunk 2]
```diff
@@ -55,10 +55,20 @@ fn print_configuration(configuration: &CliConfiguration) {
...
         "no ignore path".to_string()
     } else {
-        configuration.path_config.ignore.join(",")
+        configuration
+            .path_config
+            .ignore
+            .iter()
+            .map(|p| p.clone().into())
+            .collect::<Vec<String>>()
+            .join(",")
```
Comment (juli1) [inline crates/bins/src/bin/datadog-static-analyzer.rs:64]: Curious as to why this change? Seems more complicated to read

File: crates/cli/src/model/config_file.rs [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
 use serde;
 use serde::{Deserialize, Serialize};
 
+#[derive(Deserialize, Serialize, Debug, Default, Clone)]
+#[serde(from = "String", into = "String")]
+pub struct PathPattern {
+    pub glob: Option<GlobMatcher>,
+    pub prefix: PathBuf,
+}
+
```
Comment (juli1) [inline crates/cli/src/model/config_file.rs:17]: Can we document the use of this `struct`? A comment to explain what is it used for?

File: crates/bins/src/bin/datadog-static-analyzer.rs (hunk 1 above)
Comment (jacobotb) [reply crates/bins/src/bin/datadog-static-analyzer.rs]: Because that was the way I knew how to coerce a vector of PathPattern into a vector of String. Now I used some Rust 🪄 to solve it.

File: crates/cli/src/model/config_file.rs (hunk 3 above)
Comment (jacobotb) [reply crates/cli/src/model/config_file.rs:20]: Done.

File: crates/cli/src/file_utils.rs [hunk 4]
```diff
@@ -154,18 +153,19 @@ pub fn get_files(
...
-        return true;
-    }
-    return Path::new(path).starts_with(Path::new(pattern));
+fn matches_pattern(pattern: &PathPattern, path: &str) -> bool {
+    pattern
+        .glob
+        .as_ref()
+        .map(|g| g.is_match(path))
+        .unwrap_or(false)
+        || Path::new(path).starts_with(&pattern.prefix)
```
Comment (juli1) [inline crates/cli/src/file_utils.rs:162]: Can we implement this as part of the `PathPattern` struct?

```
impl PathPattern {
   fn match(path: &str) {
      ....
   }
}

This way, we made sure we tie/attach this function to each type? This occurred to me after reading the clarification about the `PathPattern`, it seems this function would be part of the "type".

**Review by juli1 — APPROVED** [review]

**Comment by juli1** [comment]
Approved and agreed on a follow up to make the function as part of a type.
